### ISSUE TITLE
team name mismtach: beware of complex formatting

### ISSUE THREAD

**Issue by sergpolly** [linyows/octopass#28]
Hi! Thanks for the `octopass` - looks like it's a great idea!
However I could not make `octopass` to work on either of our machines at all ...
For example, `sudo octopass passwd` gives me nothing ...

 - I've setup `/etc/octopass.conf`(see config below) and successfully installed `octopass`  - but could not make it any further.
 - Do I understand it correctly that `octopass passwd` should work even before configuring `sshd_config`, `pam.d/sshd`, `nsswitch.conf`  etc. ? it is just a GitHub API response, right ?
 - I wonder, if I should configure our github organization in particular way ? 
 - I generate a `read:org` token from my personal `github-page` - just like you described in a documentation (see picture)...
![screenshot from 2018-09-28 15-26-30](https://user-images.githubusercontent.com/6790270/46229255-fde19280-c332-11e8-85e8-f77f4d9e5079.png)
 - at the same time, `/var/cache/octopass/` is not empty - so there is some communication with github:
![screenshot from 2018-09-28 15-37-45](https://user-images.githubusercontent.com/6790270/46229730-94628380-c334-11e8-8411-3bb55a79e8b2.png)
 - I tried configuring `sshd_config`, `pam.d/sshd`, `nsswitch.conf` etc - but nothing works since `octopass` cannot see any team-members including myself (I think so).  Also configuring `nsswitch.conf` breaks the system startup - it just hang while loading (fixable using bootable usb)...
 - see al of the configs/logs below.
 - another, detail - most of our servers are behind a firewall, i.,e. they can access the web - but not the other way around - is this might be an issue ?

Any help would be greatly appreciated!

PS:
We are a small scientific lab and I was trying to use `octopass` to manage users of our couple of computational servers more efficiently. There are about ~20 of us and we have ~3-4 computers that people need to ssh into - so `octopass` seems like an ideal solution!

Here is one of the systems I've tried `octopass` on:
```
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.3 LTS
Release:        16.04
Codename:       xenial
Linux old-dell 4.10.0-42-generic #46~16.04.1-Ubuntu SMP Mon Dec 4 15:57:59 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```

Here is our `/etc/octopass.conf`
```
user@somehost:/etc$ ls -lah octopass.conf 
-rw-r--r-- 1 root root 551 Sep 28 14:13 octopass.conf
```
```
# O C T O P A S S

# Required
Token           = "108c5-----------------------------------a6d4c"

## Use team
Organization    = "dekkerlab"
Team            = "d---------------kjh"

## Use collaborators
#Owner           = "yourname"
#Repository      = "yourrepository"

# Default
#Endpoint        = "https://api.github.com/"
#Group           = "yourgroup"
Home            = "/home/%s"
Shell           = "/bin/bash"
#UidStarts       = 2000
#Gid             = 2000
#Cache           = 300
Syslog          = true

# Advanced
#SharedUsers     = [ "admin", "deploy" ]
```

Here is a snippet of the `/var/log/syslog` with some of the `octopass` messages in it:
```
Sep 28 14:26:41 old-dell systemd[8792]: Reached target Default.
Sep 28 14:26:41 old-dell systemd[8792]: Startup finished in 14ms.
Sep 28 14:26:41 old-dell systemd[1]: Started User Manager for UID 1005.
Sep 28 14:26:46 old-dell octopass[8869]: config {endpoint: https://api.github.com/, token: 108c5 ************ REDACTED ************, organization: dekkerlab, team: dekkerlab-main, owner: dekkerlab, repository: , permission:  syslog: 1, uid_starts: 2000, gid: 2000, group_name: dekkerlab-main, home: /home/%s, shell: /bin/bash, cache: 500}
Sep 28 14:26:46 old-dell octopass[8869]: _nss_octopass_setpwent_locked[L75] -- stayopen: 0
Sep 28 14:26:46 old-dell octopass[8869]: http get -- https://api.github.com/orgs/dekkerlab/teams?per_page=100
Sep 28 14:26:46 old-dell octopass[8869]: http status: 200 -- 1711 bytes retrieved
Sep 28 14:26:52 old-dell octopass[8896]: config {endpoint: https://api.github.com/, token: 108c5 ************ REDACTED ************, organization: dekkerlab, team: dekkerlab-main, owner: dekkerlab, repository: , permission:  syslog: 1, uid_starts: 2000, gid: 2000, group_name: dekkerlab-main, home: /home/%s, shell: /bin/bash, cache: 500}
Sep 28 14:26:52 old-dell octopass[8896]: _nss_octopass_setpwent_locked[L75] -- stayopen: 0
Sep 28 14:26:52 old-dell octopass[8896]: http get -- https://api.github.com/orgs/dekkerlab/teams?per_page=100
Sep 28 14:26:52 old-dell octopass[8896]: http status: 200 -- 1711 bytes retrieved
Sep 28 14:26:52 old-dell octopass[8896]: _nss_octopass_setpwent_locked[L82] -- status: UNAVAIL
Sep 28 14:26:54 old-dell octopass[8907]: config {endpoint: https://api.github.com/, token: 108c5 ************ REDACTED ************, organization: dekkerlab, team: dekkerlab-main, owner: dekkerlab, repository: , permission:  syslog: 1, uid_starts: 2000, gid: 2000, group_name: dekkerlab-main, home: /home/%s, shell: /bin/bash, cache: 500}
Sep 28 14:26:54 old-dell octopass[8907]: _nss_octopass_setpwent_locked[L75] -- stayopen: 0
Sep 28 14:26:54 old-dell octopass[8907]: use cache: /var/cache/octopass/https%3A%2F%2Fapi.github.com%2Forgs%2Fdekkerlab%2Fteams%3Fper_page%3D100-108c50
Sep 28 14:26:54 old-dell octopass[8907]: _nss_octopass_setpwent_locked[L82] -- status: UNAVAIL
Sep 28 14:26:55 old-dell octopass[8917]: config {endpoint: https://api.github.com/, token: 108c5 ************ REDACTED ************, organization: dekkerlab, team: dekkerlab-main, owner: dekkerlab, repository: , permission:  syslog: 1, uid_starts: 2000, gid: 2000, group_name: dekkerlab-main, home: /home/%s, shell: /bin/bash, cache: 500}
Sep 28 14:26:55 old-dell octopass[8917]: _nss_octopass_setpwent_locked[L75] -- stayopen: 0
Sep 28 14:26:55 old-dell octopass[8917]: use cache: /var/cache/octopass/https%3A%2F%2Fapi.github.com%2Forgs%2Fdekkerlab%2Fteams%3Fper_page%3D100-108c50
Sep 28 14:26:55 old-dell octopass[8917]: _nss_octopass_setpwent_locked[L82] -- status: UNAVAIL
Sep 28 14:26:59 old-dell octopass[8935]: config {endpoint: https://api.github.com/, token: 108c5 ************ REDACTED ************, organization: dekkerlab, team: dekkerlab-main, owner: dekkerlab, repository: , permission:  syslog: 1, uid_starts: 2000, gid: 2000, group_name: dekkerlab-main, home: /home/%s, shell: /bin/bash, cache: 500}
```

**Comment by sergpolly** [comment]
I was just manually trying to `curl` things from github api using one of my tokens - and everyhting seems to work:
curl -H "Authorization: token 108c---------------------------" https://api.github.com/teams/{team_number}/members | less```
```
this would give me a list `jsons` with actual user-info for every user in a team...

At least i'm sure now, that my token is fine.

**Comment by linyows** [comment]
@sergpolly Thank you for using octopass. It seems that you do not get a team member list after getting a list of team. What is in the cache file?

**Comment by sergpolly** [comment]
Hi!
Thanks for quick reply!
So, here is the "teams"-metadata, looks corrects to me: `https%3A%2F%2Fapi.github.com%2Forgs%2Fdekkerlab%2Fteams%3Fper_page%3D100-108c50`
I masked some numbers with `XXXXXX`:
```
[{"name":"Owners","id":XXXX,"node_id":"XXXXXXXXX=","slug":"owners","description":null,"privacy":"secret","url":"https://api.github.com/teams/XXXXXX","members_url":"https://api.github.com/teams/XXXXX/members{/member}","repositories_url":"https://api.github.com/teams/XXXXX/repos","permission":"admin"},{"name":"ATACpipe","id":XXXXX,"node_id":"XXXXXXXX=","slug":"atacpipe","description":"ATACpipe contributors","privacy":"secret","url":"https://api.github.com/teams/XXXXXX","members_url":"https://api.github.com/teams/XXXXXXX/members{/member}","repositories_url":"https://api.github.com/teams/XXXXXXX/repos","permission":"pull"},{"name":"chaos_engine","id":XXXXX,"node_id":"XXXXXXX=","slug":"chaos_engine","description":"Meta Data Repository","privacy":"secret","url":"https://api.github.com/teams/XXXXXXXX","members_url":"https://api.github.com/teams/XXXXXXXX/members{/member}","repositories_url":"https://api.github.com/teams/XXXXXXXX/repos","permission":"pull"},{"name":"chimera-tie","id":XXXXXXX,"node_id":"XXXXXXX=","slug":"chimera-tie","description":"chimera-tie contributors","privacy":"secret","url":"https://api.github.com/teams/XXXXXXXXXX","members_url":"https://api.github.com/teams/XXXXXXXXX/members{/member}","repositories_url":"https://api.github.com/teams/XXXXXXXX/repos","permission":"pull"},{"name":"DekkerLab-Main","id":XXXXXXX,"node_id":"XXXXXXXXX=","slug":"dekkerlab-main","description":"All Members of Dekker Lab","privacy":"closed","url":"https://api.github.com/teams/XXXXXXXXXX","members_url":"https://api.github.com/teams/XXXXXXX/members{/member}","repositories_url":"https://api.github.com/teams/XXXXXXXX/repos","permission":"pull"}]
```
And here is mine :
`https%3A%2F%2Fapi.github.com%2Fusers%2Fsergpolly%2Fkeys%3Fper_page%3D100-108c50` ...
oops - just realized it is empty: `[]` ...
Is it because neither of us has `ssh` keys setup with github - can this be the reason ???
I'll try to set up one and try again ...

**Comment by sergpolly** [comment]
So, i've added a bunch of ssh keys to my github - and `octopass` can retrieve those successfully now.
But `sudo octopass passwd` still does not work:
Here is a `syslog` with 3 `octopass` calls: 
1) successfull `sudo octopass {my_username}` - yields ssh keys
2) empty `sudo octopass passwd` without `/etc/nasswitch.conf` reconfig
3) empty `sudo octopass passwd` with `/etc/nasswitch.conf` reconfigured to use `octopass`
(`passwd:     files octopass sss
shadow:     files octopass sss
group:      files octopass sss`)

```
Oct  1 14:42:07 vangogh octopass[3857]: _nss_octopass_setpwent_locked[L75] -- stayopen: 0
Oct  1 14:42:07 vangogh octopass[3857]: http get -- https://api.github.com/orgs/dekkerlab/teams?per_page=100
Oct  1 14:42:07 vangogh octopass[3857]: http status: 200 -- 1711 bytes retrieved
Oct  1 14:42:07 vangogh octopass[3857]: _nss_octopass_setpwent_locked[L82] -- status: UNAVAIL
Oct  1 14:42:55 vangogh octopass[3875]: config {endpoint: https://api.github.com/, token: 32062 ************ REDACTED ************, organization: dekkerlab, team: dekkerlab-main, owner: dekkerlab, repository: , permission:  syslog: 1, uid_starts: 2000, gid: 2000, group_name: dekkerlab-main, home: /home/%s, shell: /bin/bash, cache: 500}
Oct  1 14:42:55 vangogh octopass[3875]: _nss_octopass_setgrent_locked[L68] -- stayopen: -1654104431
Oct  1 14:42:55 vangogh octopass[3875]: use cache: /var/cache/octopass/https%3A%2F%2Fapi.github.com%2Forgs%2Fdekkerlab%2Fteams%3Fper_page%3D100-32062d
Oct  1 14:42:55 vangogh octopass[3875]: _nss_octopass_setgrent_locked[L75] -- status: UNAVAIL
Oct  1 14:42:58 vangogh octopass[3881]: config {endpoint: https://api.github.com/, token: 32062 ************ REDACTED ************, organization: dekkerlab, team: dekkerlab-main, owner: dekkerlab, repository: , permission:  syslog: 1, uid_starts: 2000, gid: 2000, group_name: dekkerlab-main, home: /home/%s, shell: /bin/bash, cache: 500}
Oct  1 14:42:58 vangogh octopass[3881]: _nss_octopass_setgrent_locked[L68] -- stayopen: -173543791
Oct  1 14:42:58 vangogh octopass[3881]: http get -- https://api.github.com/orgs/dekkerlab/teams?per_page=100
Oct  1 14:42:58 vangogh octopass[3881]: http status: 200 -- 1711 bytes retrieved
Oct  1 14:42:58 vangogh octopass[3881]: _nss_octopass_setgrent_locked[L75] -- status: UNAVAIL
Oct  1 14:42:58 vangogh octopass[3883]: config {endpoint: https://api.github.com/, token: 32062 ************ REDACTED ************, organization: dekkerlab, team: dekkerlab-main, owner: dekkerlab, repository: , permission:  syslog: 1, uid_starts: 2000, gid: 2000, group_name: dekkerlab-main, home: /home/%s, shell: /bin/bash, cache: 500}
Oct  1 14:42:58 vangogh octopass[3883]: _nss_octopass_setpwent_locked[L75] -- stayopen: 0
Oct  1 14:42:58 vangogh octopass[3883]: use cache: /var/cache/octopass/https%3A%2F%2Fapi.github.com%2Forgs%2Fdekkerlab%2Fteams%3Fper_page%3D100-32062d
Oct  1 14:42:58 vangogh octopass[3883]: _nss_octopass_setpwent_locked[L82] -- status: UNAVAIL

```

**Comment by sergpolly** [comment]
Alright, I think I found something:
```{"name":"DekkerLab-Main","id":xxxxxx,"node_id":"xxxxxxxx","slug":"dekkerlab-main", THE REST DOES NOT MATTER ...}```
Turned out - I was trying to refer to the `team` by its `slug` - i.e. lower-cased version of its `name` - which had more complicated formatting ...
I get output from `sudo octopass passwd` now!
It would great if  `octopass` printed an error message about non-existing `team ` somewhere around here:
https://github.com/sergpolly/octopass/blob/d80fca69654aa0b29268827d704ac7fc4c5db089/octopass.c#L473

**Comment by linyows** [comment]
Great! I see. When I coded octopass, slug probably did not exist in the github team. I thought that it was better to search for slug in addition to name, and I thought that you should output errors that the team can not find for debug, as you say. I will make these support soon.

**Comment by linyows** [comment]
I released as v0.5.0.

**Comment by sergpolly** [comment]
Awesome! Thanks a lot!
